### PR TITLE
[.NET] Dutch DateTime patial support

### PR DESCRIPTION
**Pull request by LionbridgeCS**
Work in progress supporting  Dutch DateTime . 

DateTimeExtractor | 116 Passing out of 217
DateParser | 112 Passing out of  210
DatePeriodExtractor | 164 Passing out of 303
DateTimeExtractor | 43 Passing out of 128
DateTimeParser | 14 Passing out of 42
DateTimePeriodExtractor | 29 Passing out of 79
DateTimePeriodParser | 32 Passing out of  156
DurationExtractor | 32 Passing out of  85
DurationParser | 24 Passing out of 76
HolidayExtractor | 15 Passing out of  21
HolidayParser | 17 Passing out of 25
MergedExtractor | 44 Passing out of  73
SetExtractor | 12 Passing out of  48
TimeExtractor | 30 Passing out of 96
TimeParser | 25 Passing out of  135
TimePeriodExtractor | 48 Passing out of  119
TimePeriodParser | 30 Passing out of  59

****************************************************************
test cases to be marked as skipped later.

### CODE REVIEW

File: Patterns/Dutch/Dutch-DateTime.yaml [hunk 1]
```diff
@@ -614,30 +610,31 @@ TimeBeforeAfterRegex: !nestedRegex
...
 ComplexDatePeriodRegex: !nestedRegex
-  def: (((from|during|in)\s+)?(?<start>.+)\s*({TillRegex})\s*(?<end>.+)|((between)\s+)(?<start>.+)\s*({RangeConnectorRegex})\s*(?<end>.+))
+  def: (((vanaf|tijdens|gedurende|in)\s+)?(?<start>.+)\s*({TillRegex})\s*(?<end>.+)|((tussen)\s+)(?<start>.+)\s*({RangeConnectorRegex})\s*(?<end>.+))
   references: [ YearRegex, TillRegex, RangeConnectorRegex ]
 UnitMap: !dictionary
   types: [ string, string ]
   entries:
-    millennium: 1000Y
-    eeuw: 100Y
+    millennium: 1000M
```
Comment (iMicknl) [inline Patterns/Dutch/Dutch-DateTime.yaml]: Same feedback as previous PR. `1000M` should be `1000Y`, keys should not be translated as they are only used internally.

File: Patterns/Dutch/Dutch-DateTime.yaml [hunk 2]
```diff
@@ -614,30 +610,31 @@ TimeBeforeAfterRegex: !nestedRegex
...
+    eeuw: 100M
```
Comment (iMicknl) [inline Patterns/Dutch/Dutch-DateTime.yaml]: Same feedback.

**Comment by iMicknl** [comment]
Is this PR a follow up of #1665? Has the feedback of #1665 been addressed in this one or does this one contain different functionality?

**Comment by LionbridgeCS** [comment]
@iMicknl this is a follow up  of #1665. it is a WIP and will be updated as the implementation continues. 
This one will include all the DateTime type. #1665 included only the tests for Date sub-types. the other sub-types will be broken down too.  
Some of the feedback was already handled here and more of it will be handled in the updates.

File: Patterns/Dutch/Dutch-DateTime.yaml [hunk 3]
```diff
@@ -1,44 +1,44 @@
...
 AfterNextSuffixRegex: !simpleRegex
-  def: \b(na\s+(de\s+)?volgende)\b
+  def: \b(na\s+((de|het)\s+)?volgende)\b
 PastPrefixRegex: !simpleRegex
-  def: .^
-# TODO: modify below regex according to the counterpart in English
+  def: ((deze\s+)?verleden)\b
 PreviousPrefixRegex: !nestedRegex
-  def: (voorgaande|vorig|verleden|vorige|laatste|{PastPrefixRegex})\b
+  def: (voorgaand[e]|vorig[e]?|afgelopen|verleden|laatste|{PastPrefixRegex})\b
```
Comment (iMicknl) [inline Patterns/Dutch/Dutch-DateTime.yaml]: `e` in `voorgaand[e]` should be optional? Like `voorgaand[e]?`

File: Patterns/Dutch/Dutch-DateTime.yaml [hunk 4]
```diff
@@ -1,44 +1,44 @@
...
-RangeConnectorRegex : !nestedRegex
-  def: (?<and>\b(and|through|to)\b|{BaseDateTime.RangeConnectorSymbolRegex})
+RangeConnectorRegex: !nestedRegex
+  def: (?<and>\b(en|tot en met|t\\m|tot|tot aan)\b|{BaseDateTime.RangeConnectorSymbolRegex})
   references: [ BaseDateTime.RangeConnectorSymbolRegex ]
+ArticleRegex: !simpleRegex
+  def: \b(de|het|een)\b
 RelativeRegex: !simpleRegex
-  def: \b(?<order>dit|volgende|komende|aankomende|aanstaande|deze|huidige|vorig|verleden|voorgaande|vorige|laatste)\b
+  def: \b(?<order>dit|deze|volgend[e]?|komend[e]?|aankomend[e]?|aanstaand[e]?|huidig[e]?|vorig[e]?|verleden|voorgaand[e]?|vorig[e]?|laatste|afgelopen)\b
```
Comment (iMicknl) [inline Patterns/Dutch/Dutch-DateTime.yaml]: @tellarin @STayinloves any preference between using `(e)?` vs `[e]?`? When it applies to a single character, like the example above. Afaik performance and functionality is the same.

File: Patterns/Dutch/Dutch-DateTime.yaml [hunk 5]
```diff
@@ -1067,13 +1078,15 @@ SuperfluousWordList: !list
     - maybe
     - say
     - like
-DurationDateRestrictions: [ vandaag, nu ]
 # For DurationDateRestrictions, only translate these terms if this particular circumstance applies in the target language. If not, keep it empty. 
+DurationDateRestrictions: [ today, now ]
```
Comment (iMicknl) [inline Patterns/Dutch/Dutch-DateTime.yaml]: _For DurationDateRestrictions, only translate these terms if this particular circumstance applies in the target language. If not, keep it empty._ 

If this doesn't apply, keep them empty. Otherwise revert the changes in translation.

File: Patterns/Dutch/Dutch-DateTime.yaml (hunk 4 above)
Comment (STayinloves) [reply Patterns/Dutch/Dutch-DateTime.yaml]: `(e)` catch a group while `[e]` doesn't. I think just `e?` is sufficient.

File: Patterns/Dutch/Dutch-DateTime.yaml (hunk 3 above)
Comment (LionbridgeCS) [reply Patterns/Dutch/Dutch-DateTime.yaml]: yes, this will be updated to be optional. 
"voorgaand" is the uninflected form, "voorgaande" the inflected (and most usual) form.  "voorgaande" as default but "voorgaand"  is also used.

File: Patterns/Dutch/Dutch-DateTime.yaml (hunk 5 above)
Comment (LionbridgeCS) [reply Patterns/Dutch/Dutch-DateTime.yaml]: @iMicknl & @STayinloves: question, if there is a regex from English that is not used in Dutch or other languages should it be removed from the respective yaml?

**Comment by LionbridgeCS** [comment]
@iMicknl & @STayinloves: question, if there is a regex from English that is not used in Dutch or other languages should it be removed from the respective yaml?

File: Patterns/Dutch/Dutch-DateTime.yaml (hunk 5 above)
Comment (LionbridgeCS) [reply Patterns/Dutch/Dutch-DateTime.yaml]: @iMicknl so this should be changed to DurationDateRestrictions: []

Comment (iMicknl) [reply Patterns/Dutch/Dutch-DateTime.yaml]: Yes 👍

**Comment by iMicknl** [comment]
> @iMicknl & @STayinloves: question, if there is a regex from English that is not used in Dutch or other languages should it be removed from the respective yaml?

Which regexes are involved?
My preference would be to keep them for now, make all test pass and then clean up. Before some regexes are removed, which will be needed later on for full support.
Agree @STayinloves?

**Comment by LionbridgeCS** [comment]
@iMicknl thank you for the feedback! this was more of a general question :)

**Comment by STayinloves** [comment]
> > @iMicknl & @STayinloves: question, if there is a regex from English that is not used in Dutch or other languages should it be removed from the respective yaml?
> 
> Which regexes are involved?
> My preference would be to keep them for now, make all test pass and then clean up. Before some regexes are removed, which will be needed later on for full support.
> Agree @STayinloves?

It depends. Most regexes are used to configure some interface, in this case, it can't be removed but should be assigned specific values (e.g. non-match value `[^.]`). But if it acts just a dict/list item. it can be removed after testing.

**Comment by LionbridgeCS** [comment]
@tellarin, should this one be closed as the feedback from here was handled both in here and in #1665

**Comment by tellarin** [comment]
I think it's OK to close this, or to merge the contents of #1665 + #1676.
